packages/synapse-react-client/src/lib/containers/useImmutableTableQuery.ts

```diff
@@ -194,6 +195,11 @@ export default function useImmutableTableQuery(
     setQuery(initQueryRequest)
   }, [initQueryRequest, setQuery])
 
+  /* If the initial query changes, then reset the query to match the new prop */
+  useDeepCompareEffect(() => {
+    resetQuery()
+  }, [initQueryRequest])
+
```

Comment:
The issue occurred because `initQueryRequest` was only used to initialize state. Any updates to this prop would not trigger a re-render. In this hook, let's track the object for changes and re-render the component if it does change.

We will use `useDeepCompareEffect` because otherwise we would need to memoize this object in each component where this hook is used.